src/DI/SlackLoggerExtension.php

```diff
@@ -16,41 +16,45 @@
 
 class SlackLoggerExtension extends CompilerExtension
 {
-	private $defaults = [
-		'enabled' => FALSE,
-		'logUrl' => NULL,
-		'channel' => NULL,
-		'username' => NULL,
-		'icon' => NULL,
-		'pretext' => NULL,
-	];
-
-
-	public function afterCompile(ClassType $class)
-	{
-		$config = $this->getConfig($this->defaults);
-
-		Validators::assertField($config, 'enabled', 'boolean');
-
-		if ($config['enabled']) {
-			Validators::assertField($config, 'slackUrl', 'string');
-			Validators::assertField($config, 'logUrl', 'string|null');
-			Validators::assertField($config, 'channel', 'string|null');
-			Validators::assertField($config, 'username', 'string|null');
-			Validators::assertField($config, 'icon', 'string|null');
-			Validators::assertField($config, 'pretext', 'string|null');
-
-			$init = $class->getMethod('initialize');
-			$init->addBody('?::setLogger(new ?(?, ?, ?, ?, ?, ?));', [
-				new PhpLiteral(Debugger::class),
-				new PhpLiteral(SlackLogger::class),
-				$config['slackUrl'],
-				$config['logUrl'],
-				$config['channel'],
-				$config['username'],
-				$config['icon'],
-				$config['pretext'],
-			]);
-		}
-	}
+    private $defaults = [
+        'enabled' => FALSE,
+        'logUrl' => NULL,
+        'channel' => NULL,
+        'username' => NULL,
+        'icon' => NULL,
+        'pretext' => NULL,
+    ];
+
+
+    public function afterCompile(ClassType $class)
+    {
+        $config = $this->getConfig($this->defaults);
+
+        Validators::assertField($config, 'enabled', 'boolean');
+
+        if ($config['enabled']) {
+            Validators::assertField($config, 'slackUrl', 'string');
+            Validators::assertField($config, 'logUrl', 'string|null');
+            Validators::assertField($config, 'channel', 'string|null');
+            Validators::assertField($config, 'username', 'string|null');
+            Validators::assertField($config, 'icon', 'string|null');
+            Validators::assertField($config, 'pretext', 'string|null');
+            Validators::assertField($config, 'file', 'string|null');
+            Validators::assertField($config, 'interval', 'int|null');
```

Comment:
This says `$interval` shall be an int, while `SlackLogger::$interval` PhpDoc says it may be a string, too. Please, unify it, and document somewhere (in the PhpDoc, perhaps) what the value may be.